spec/examples/schema_adapter_examples.rb

```diff
@@ -142,10 +146,28 @@
     end
     # rubocop:enable RSpec/MultipleExpectations
 
-    it 'allows a list of schemas' do
-      subject.switch([schema1, schema2]) do
-        expect(connection.schema_search_path).to include %("#{schema1}")
-        expect(connection.schema_search_path).to include %("#{schema2}")
+    describe 'multiple schemas' do
+      it 'allows a list of schemas' do
+        subject.switch([schema1, schema2]) do
+          expect(connection.schema_search_path).to include %("#{schema1}")
+          expect(connection.schema_search_path).to include %("#{schema2}")
+        end
+      end
+
+      it 'connects and resets' do
+        subject.switch([schema1, schema2]) do
+          # Ensure sequence is not cached
+          Company.reset_sequence_name
+          User.reset_sequence_name
```

Comment:
Without resetting the sequence, the spec would pass if ran with the higher level spec with the same name because the sequence_name is cached on the model. I applied the same process to the other spec to avoid false positives depending on how RSpec orders the tests.

Let me know if there is a better way to do this.